Add docs for artifactory agent backfill mode (#17112)
docker/docs: scout/artifactory.md

  docker/artifactory-agent:v1
```

#### Analyzing pre-existing data

> By default the agent will detect and analyze images as they're created and

Reviewer: ```suggestion
By default the agent detects and analyzes images as they're created and
```

> updated. If you would like to use the agent to analyze pre-existing images, you

Reviewer: ```suggestion
updated. If you want to use the agent to analyze pre-existing images, you
```

> can use backfill mode. Use the `--backfill-from=TIME` command line argument,

Reviewer: ```suggestion
can use backfill mode. Use the `--backfill-from=TIME` command line option,
```

@jonnystoten We haven't mentioned this anywhere, but standardising the language we use around command line "things" came up in a conversation elsewhere and hasn't been recorded yet ;)

where `TIME` is an ISO 8601 formatted time, to run the agent in backfill mode.

> If this argument is given, the agent analyzes all images pushed between that

Reviewer: ```suggestion
If you use this option, the agent analyzes all images pushed between that
```

time and the current time when the agent starts, then exits.

For example:

```console
$ docker run \
  --mount type=bind,src=/var/opt/artifactory-agent,target=/opt/artifactory-agent/data \
  docker/artifactory-agent:v1 --backfill-from=2022-04-10T10:00:00Z
```

When running a backfill multiple times, the agent won't analyze images that

> were already analyzed. To force re-analysis, provide the `--force` command

Reviewer: ```suggestion
it's already analyzed. To force re-analysis, provide the `--force` command
```

> line argument.

Reviewer: ```suggestion
line flag.
```

this is the other side of that conversation… option vs flag…
